### PR TITLE
Extract scripts from xip patch

### PR DESCRIPTION
**Pull request by sjanota**
**Description**

Changes proposed in this pull request:

- Extract get-service-ip.sh script
- Extract generate-cert.sh script

**Related issue(s)**

#1989

### CODE REVIEW

File: components/xip-patch/generate-cert.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
```
Comment (strekm) [inline components/xip-patch/generate-cert.sh]: add `set -o errexit`

File: components/xip-patch/get-service-ip.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
```
Comment (strekm) [inline components/xip-patch/get-service-ip.sh]: add `set -o errexit`

File: components/xip-patch/xip-patch.sh [hunk 3]
```diff
@@ -2,48 +2,21 @@
...
+DOMAIN="${EXTERNAL_PUBLIC_IP}.xip.io"
 
-openssl req -x509 -nodes -days 5 -newkey rsa:4069 \
-                 -subj "/CN=${PUBLIC_DOMAIN}" \
-                 -reqexts SAN -extensions SAN \
-                 -config <(cat /etc/ssl/openssl.cnf \
-        <(printf "\\n[SAN]\\nsubjectAltName=DNS:*.%s" "${PUBLIC_DOMAIN}")) \
-                 -keyout "${KEY_PATH}" \
-                 -out "${CERT_PATH}"
+OUT_DIR=${DIR} ${DIR}/generate-cert.sh
```
Comment (strekm) [inline components/xip-patch/xip-patch.sh]: `${DOMAIN}`, `${KEY_PATH}` and `${CERT_PATH}` are not exported

File: components/xip-patch/xip-patch.sh [hunk 4]
```diff
@@ -2,48 +2,21 @@
...
 
 TLS_CERT=$(base64 "${CERT_PATH}" | tr -d '\n')
```
Comment (strekm) [inline components/xip-patch/xip-patch.sh]: `${CERT_PATH}` not declared in this script

File: components/xip-patch/xip-patch.sh [hunk 5]
```diff
@@ -2,48 +2,21 @@
...
 TLS_KEY=$(base64 "${KEY_PATH}" | tr -d '\n')
```
Comment (strekm) [inline components/xip-patch/xip-patch.sh]: `${KEY_PATH}` not declared in this script

**Comment by sjanota** [comment]
Already on master thanks to #2793

**Pull request closed without merging**